Develop

AndreyRogozhin opened the pull request.

Commit:
Sprint_6

Commit:
add jacoco report

nikolay-977 commented on src/main/java/com/example/Lion.java:
```diff
@@ -5,18 +5,19 @@
 public class Lion {
 
     boolean hasMane;
```
Можно улучшить: Для всех полей следует указывать модификаторы доступа.

AndreyRogozhin replied:
Исправлено

nikolay-977 commented on src/test/java/com/example/TestFelineKittens.java:
```diff
@@ -0,0 +1,56 @@
+
+    @Test
+    public void runTestGetKittens() {
+        if (kittensExpected == -1) {
```
Необходимо исправить: Не следует использовать конструкцию if-else при написании тестов. Это затрудняет читаемость кода и может вызвать путаницу в понимании. Лучше реализовать 2 отдельных кейса

AndreyRogozhin replied:
Исправлено. Тесты разнесены в 2 класса.

nikolay-977 reviewed (commented):
Можно улучшить: хорошей практикой является добавление README.md с описанием проекта, которое включает название, технологии в проекте (например, Java 11, JUnit 4.13.2, maven 3.9.0 и т.д.), настройка (если необходима) и как запускать (`mvn clean test` - кавычки позволяют выполнять запуск из файла README.md). Как оформлять README.md можно ознакомиться в статье https://blog.jetbrains.com/idea/2022/08/markdown-support-improvements/

nikolay-977 commented on src/test/java/com/example/TestFeline.java:
```diff
@@ -0,0 +1,45 @@
+
+    }
+
+
```
Можно улучшить: лишних пропусков строк следует избегать. Для зрительного разделения кода достаточно использовать один пропуск строки. Пропуски строк должны быть однообразными.
https://www.oracle.com/java/technologies/javase/codeconventions-whitespace.html
https://www.magnumblog.space/java/translating-java-code-conventions#blank_lines

AndreyRogozhin replied:
README.md дополнен

nikolay-977 reviewed (commented):
Можно улучшить: хорошей практикой является добавление README.md с описанием проекта, которое включает название, технологии в проекте (например, Java 11, JUnit 4.13.2, maven 3.9.0 и т.д.), настройка (если необходима) и как запускать (`mvn clean test` - кавычки позволяют выполнять запуск из файла README.md). Как оформлять README.md можно ознакомиться в статье https://blog.jetbrains.com/idea/2022/08/markdown-support-improvements/

Commit:
Sprint_6

Дополнен файл README.md.
Удалено ветвление  if-else
Удалены лишние пустые строки
Указан модификатор доступа

Commit:
Sprint_6

Дополнен файл README.md.
Удалено ветвление  if-else
Удалены лишние пустые строки
Указан модификатор доступа
